### PR TITLE
Speculative fixes for intermittent test in DeleteAndQuitTest

### PR DESCRIPTION
**Pull request by jonalmeida**
For https://github.com/mozilla-mobile/fenix/pull/20782#issuecomment-897311109: `advanceUntilIdle` might solve for the nested coroutines without needing to change the implementation for now.

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not

### CODE REVIEW

File: app/src/test/java/org/mozilla/fenix/settings/deletebrowsingdata/DeleteAndQuitTest.kt [hunk 1]
```diff
@@ -94,44 +102,52 @@ class DeleteAndQuitTest {
         }
     }
 
-    @Ignore("Intermittently failing; will be fixed with #5406.")
+    @Ignore("Failing test; need more investigation.")
```
Comment (eliserichards) [inline app/src/test/java/org/mozilla/fenix/settings/deletebrowsingdata/DeleteAndQuitTest.kt:105]: This one is still failing when I run it and remove this @ignore :/

Comment (jonalmeida) [reply app/src/test/java/org/mozilla/fenix/settings/deletebrowsingdata/DeleteAndQuitTest.kt:105]: Yeah, I updated the test to follow current implementation and left a comment for now, but I don't have the bandwidth to debug all the tests.

**Comment by jonalmeida** [comment]
Restarting to see if we have any intermittent..

**Comment by jonalmeida** [comment]
`mainMenuInstallPWATest` UI test failure is unrelated to this change. Let's try again to get another round of green ✅ .

**Review by Amejia481 — APPROVED** [review]

**Comment by jonalmeida** [comment]
The UI tests that are failing here have been disabled on main.

**Comment by jonalmeida** [comment]
Welp, test is still failing. 🤦 
```
delete only tabs and quit

java.lang.AssertionError: Verification failed: fewer calls happened than demanded by order verification sequence. 

Matchers: 
FenixSnackbar(#507).show())
RemoveAllTabsUseCase(#506).invoke(eq(false)))
HomeActivity(#499).finishAndRemoveTask())

Calls:
1) HomeActivity(#499).equals(Context(#11))

Stack traces:
1)                                                              io.mockk.impl.InternalPlatform.captureStackTrace                 (InternalPlatform.kt:124)                  
                                                                  io.mockk.impl.stub.MockKStub.handleInvocation                  (MockKStub.kt:250)                         
                                io.mockk.impl.instantiation.JvmMockFactoryHelper$mockHandler$1.invocation                        (JvmMockFactoryHelper.kt:23)               
                                                         io.mockk.proxy.jvm.advice.Interceptor.call                              (Interceptor.kt:21)           
```

**Comment by jonalmeida** [comment]
I'm disabling this test for now, it's still failing despite my fixes and it's affecting other PRs too often. 😕

**Comment by Amejia481** [comment]
👍🏽

**Review by Amejia481 — APPROVED** [review]

**Pull request merged**